wkcuber/api/Properties/LayerProperties.py

```diff
@@ -161,23 +176,23 @@ def bounding_box(self) -> dict:
         return self._bounding_box
 
     @property
-    def wkw_magnifications(self) -> dict:
+    def wkw_magnifications(self) -> List[Resolution]:
         return self._wkw_magnifications
 
 
 class SegmentationLayerProperties(LayerProperties):
     def __init__(
         self,
-        name,
-        category,
-        element_class,
-        data_format,
-        num_channels,
-        bounding_box=None,
-        resolutions=None,
-        largest_segment_id=None,
-        mappings=None,
-    ):
+        name: str,
+        category: str,
+        element_class: str,
+        data_format: str,
+        num_channels: int,
+        bounding_box: Dict[str, Union[int, Tuple[int, int, int]]] = None,
+        resolutions: List[Resolution] = None,
+        largest_segment_id: int = None,
```

Comment:
I think `largest_segment_id` is not optional for segmentation layers, but making it non-optional would require a different order of the parameters and then the order of `SegmentationLayerProperties` would be different from `LayerProperties`. 
(There is an assertion in DatasetProperties._add_layer for this)
Therefore, I would leave it this way.

Reply:
Yes, I think you can leave it like this or have a default of `-1`, if that is useful? If you want to leave the signature as is, you can `assert largest_segment_id is not None` in the `__init__()` to mark the attribute as non-optional.